### PR TITLE
Weekly images

### PR DESCRIPTION
**Pull request by mvollmer**
Depends on https://github.com/cockpit-project/cockpituous/pull/19

Later:
- Maybe fill out check_publishing
- Worry about creating non-public images

### CODE REVIEW

File: test/testinfra.py [hunk 1]
```diff
@@ -401,22 +423,81 @@ def update_status(revision, context, last, changes):
...
+                                                                             "pull/%d/head" % number, context)))
+
+    def scan_for_image_tasks(self):
+        # XXX
+        return [ (10, GithubImageTask("refresh-fedora-23", "fedora-23")) ]
+
+        issues = self.get("issues?labels=bot")
+
+        results = [ ]
+        for image in DEFAULT_IMAGE_REFRESH:
```
Comment (dperpeet) [inline test/testinfra.py]: This doesn't take into consideration that an image might be set to False. If it's used like this, it should be an array or list.

File: test/testinfra.py [hunk 2]
```diff
@@ -401,22 +423,81 @@ def update_status(revision, context, last, changes):
...
+            found = False
+            for issue in issues:
+                if issue['title'] == ISSUE_TITLE_IMAGE_REFRESH.format(image):
+                    age = time.time() - time.mktime(time.strptime(issue['created_at'], "%Y-%m-%dT%H:%M:%SZ"))
+                    if age < IMAGE_REFRESH * 24 * 60 * 60:
+                        found = True
+                    print "not old enough", image, age
+            if not found:
+                results.append(GitHub.TaskEntry(10, GithubImageTask("refresh-" + image, image)))
```
Comment (dperpeet) [inline test/testinfra.py]: we should make "baseline" a variable, and let the master priority and this one depend on that
https://github.com/cockpit-project/cockpit/blob/0a95300a0abbe0ac6e1497b3ab2c573f70d50ab6/test/testinfra.py#L389

File: test/testinfra.py (hunk 1 above)
Comment (mvollmer) [inline test/testinfra.py]: I have changed the dict to `{ "fedora-23" : { }, ... }`.  The idea is that we might want to add some config properties later.

File: test/testinfra.py (hunk 2 above)
Comment (mvollmer) [inline test/testinfra.py]: Done

**Comment by mvollmer** [comment]
This should use `github.qualify` instead of `github.base`.

**Comment by mvollmer** [comment]
This should use `${comments_url}` instead of `${url/comments}`.

**Comment by mvollmer** [comment]
This needs to be adapted to the new interpolation syntax.

**Comment by mvollmer** [comment]
Something broke: https://github.com/cockpit-project/cockpit/issues/3651

**Comment by mvollmer** [comment]
> This should use github.qualify instead of github.base.

Done.

**Comment by mvollmer** [comment]
`testBasic (check_embed.TestEmbed)` seems to be broken.  I can't immediately see how the changes here could do that.

**Comment by mvollmer** [comment]
> testBasic (check_embed.TestEmbed) seems to be broken.

Fixed here: #3659.

**Comment by mvollmer** [comment]
> Something broke: #3651

The branch name `cockpitous:refresh-fedora-bla` was expanded to `cockpitousNone`.  Fixed here and in the sink.

**Comment by mvollmer** [comment]
Made with this code: https://github.com/cockpit-project/cockpit/pull/3660

**Comment by stefwalter** [comment]
This got merged into master along with #3660

**Pull request closed without merging**